### ISSUE TITLE
Get decimal part of reading

### ISSUE THREAD

**Issue by ilikegitlab** [RobTillaart/DHTINT#3]
For humidity, I could live with an integer values (the datasheet claims it has no greater repeatability than 1% anyway, though it probably depends also on the RH range we are looking at. Haven't extensively tested this sensor). It would still be nice to have a less jumpy response in some application.

I feel it is a bit unfortunate there is no way to get the decimal part of the reading as should be able to get it until 0.2 degC precise (I know, i am complaining about a few bytes because floats but don't want to give up a decimal).

 I'm having a couple of other sensors in my project and am using all integer math. For the other sensors I use two bytes per reading (one for the integer and one for the decimal part). 

I don't know how DHTint could best do it and keep the api similar as in DHTnew. Maybe a separate read method, or option to the read method?

Then again. Since you are thinking about tenth of degree offsets in issue #2 , maybe it makes sense to add a "scale" method to say we want everything in tenth of degree?

**Comment by RobTillaart** [comment]
(some thoughts)
The decimal part of the temperature and temperature could be stored in one byte each. Better might be a function to get the rounding error. 

Temp = 50.  RE = -3.  So temp was 50.3
Temp = 50   RE = 3. So temp was 49.7

Or just get the int part and decimal part. 
The values would become

50 ....3 
49.....7

The latter method gives "rounding errors" for people only using the integer part.

**Comment by RobTillaart** [comment]
It will be important to keep the interfaces of DHTNEW and DHTINT compatible as possible. The scaling proposal makes the functions ambiguous

**Comment by ilikegitlab** [comment]
Agree, scaling is probably  a bad idea. I like both RE and int and decimal part. Not sure what is best or more clear.

**Comment by RobTillaart** [comment]
Your proposal to let the user divide by 10 is still a valid option.  
It would not break the compatibility with DHTNEW if we add two functions:
- **int getTemperatuteTenth()**
- **int getHumidityTenth()**

To implement these extra code (PROGMEM) is needed, RAM can be the same. 

I'll give it a try in a develop branch to get an indication of the impact. 

----
insights
- Offset would also need 2 extra functions to be able to do in tenth (=> extra footprint)
- different formulas of the different sensors need updates.

**Comment by RobTillaart** [comment]
just to keep the thought: 
- give the user access to the raw bytes.

**Comment by RobTillaart** [comment]
pushed develop branch - only changed dhtint.cpp - with 
- temperature in one tenth,
- humidity in one tenth,
- user should divide by 10 for integer part,
- user should modulo by 10 for decimal part,
- offsets are in one tenth too

Footprint of UNO minimum sketch dropped from 3978 to 3852  => 110 bytes gained but interface broken.

----
update
- build was successful => you can give it a try - no it won't be merged into master
- it has a small optimization that could be used ( about 16 bytes gain)

**Comment by ilikegitlab** [comment]
works for me.

I would be happy with raw bytes as well (especially if that can be in master as I can't expect the users from the project to install development versions).

**Comment by RobTillaart** [comment]
I could add to the dhtint.h file the following two functions without breaking the compatibility.
```cpp

uint16_t getRawTemperature()
{
  uint16_t raw = _bits[2] * 256 + _bits[3];
  return raw;
};


uint16_t getRawHumidity()
{
  uint16_t raw = _bits[0] * 256 + _bits[1];
  return raw;
}
```
how to decode the uint16_t depends on the type of the sensor...

**Comment by RobTillaart** [comment]
Created a branch called raw => if CI build succeeds this will be the 0.1.1 release.

**Comment by RobTillaart** [comment]
Merged raw branch and released 0.1.1 version.
If this works you can close the issue.

**Comment by RobTillaart** [comment]
Found a bug working on #2 

bug - these functions introduce float math 
```cpp
  float    getHumOffset()                { return _humOffset; };
  float    getTempOffset()               { return _tempOffset; };
```

Not the most often used functions but they will cause linking the float libraries again..
(will be fixed in 0.1.2 someday)

**Comment by ilikegitlab** [comment]
using raw resolved the issue for me.

thanks!
